### PR TITLE
Fix defaults with oneof/anyof and nested dependencies

### LINKED ISSUE #768: Default values in dependency are ignored
### Prerequisites

- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme)

### Description
When using dependencies, the default values in the dependency object are ignored.

### Steps to Reproduce
```
{
    "title": "Person",
    "type": "object",
    "properties": {
        "Do you have any pets?": {
            "type": "string",
            "enum": [
                "No",
                "Yes: One",
                "Yes: More than one"
            ],
            "default": "No"
        }
    },
    "required": [
        "Do you have any pets?"
    ],
    "dependencies": {
        "Do you have any pets?": {
            "oneOf": [
                {
                    "properties": {
                        "Do you have any pets?": {
                            "enum": [
                                "No"
                            ]
                        },
                        "title" : {
                            "type": "string",
                            "title": "Title",
                            "default": "A new task"
                        }
                    }
                },
                {
                    "properties": {
                        "Do you have any pets?": {
                            "enum": [
                                "Yes: One"
                            ]
                        },
                        "How old is your pet?": {
                            "type": "number"
                        }
                    },
                    "required": [
                        "How old is your pet?"
                    ]
                },
                {
                    "properties": {
                        "Do you have any pets?": {
                            "enum": [
                                "Yes: More than one"
                            ]
                        },
                        "Do you want to get rid of any?": {
                            "type": "boolean"
                        }
                    },
                    "required": [
                        "Do you want to get rid of any?"
                    ]
                }
            ]
        }
    }
}
```
In this case I used the given Person Example and added a Title field with a default value.
But this default value is ignored. 

I figured out that the _computeDefaults_ function is never called for the elements in the dependency object.

### Version
1.0.0

### LINKED ISSUE #1229: Form fields are not populated with default values when using oneOf
### Prerequisites

- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [X] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] Ideally, I'm providing a [shared playground link](https://mozilla-services.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJkZWZpbml0aW9ucyI6eyJCYXIiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsiQ29uc3QiOnsiY29uc3QiOiJCYXIiLCJkZWZhdWx0IjoiQmFyIn0sIkJhclAiOnsidHlwZSI6ImludGVnZXIifX19LCJCYXoiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsiQ29uc3QiOnsiY29uc3QiOiJCYXoiLCJkZWZhdWx0IjoiQmF6In0sIkJhelAiOnsidHlwZSI6ImludGVnZXIifX19fSwidHlwZSI6Im9iamVjdCIsIm9uZU9mIjpbeyJ0aXRsZSI6IkJhciIsIiRyZWYiOiIjL2RlZmluaXRpb25zL0JhciJ9LHsidGl0bGUiOiJCYXoiLCIkcmVmIjoiIy9kZWZpbml0aW9ucy9CYXoifV19LCJ1aVNjaGVtYSI6e319) demonstrating the issue.

### Description

1. When using `oneOf` the form fields are not populated with the their specified default values. 
2. Changing the selected option does not update the fields with the default value.

#### Expected behavior

The fields should have their contents populated with the default value.

#### Actual behavior

The fields are empty.

### LINKED ISSUE #1293: Issue with default values in anyOf/oneOf dependencies [PG example included]
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://mozilla-services.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

As mentioned previously in issue #975, there seems to be a bug related to using default values in combination with anyOf/oneOf dependency fields. In issue #975 epicfaace asked that we provide a working playground example, which is now available below.

### Steps to Reproduce

See playground example:
[default values playground example](https://mozilla-services.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJydGMiOnsic3luYyI6MH19LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsicnRjIjp7InRpdGxlIjoiQ29uZmlndXJlIENsb2NrIiwidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsic3luYyI6eyJ0aXRsZSI6IlRpbWUgU3luYyBUeXBlIiwidHlwZSI6ImludGVnZXIiLCJkZWZhdWx0IjowLCJhbnlPZiI6W3sidGl0bGUiOiJSZXRhaW4gdGltZSIsInR5cGUiOiJpbnRlZ2VyIiwiZW51bSI6WzBdfSx7InRpdGxlIjoiQXV0byB0aW1lIiwidHlwZSI6ImludGVnZXIiLCJlbnVtIjpbMV19XX19LCJkZXBlbmRlbmNpZXMiOnsic3luYyI6eyJvbmVPZiI6W3sicHJvcGVydGllcyI6eyJzeW5jIjp7ImVudW0iOlswXX19fSx7InByb3BlcnRpZXMiOnsic3luYyI6eyJlbnVtIjpbMV19LCJ0b2xlcmFuY2UiOnsidGl0bGUiOiJUaW1lIHN5bmMgdG9sZXJhbmNlIiwidHlwZSI6ImludGVnZXIiLCJtaW5pbXVtIjoxLCJtYXhpbXVtIjozNjAwLCJkZWZhdWx0IjozMH19fV19fX19fSwidWlTY2hlbWEiOnt9fQ==)

#### Expected behavior

When the user selects the "Auto time" field in the dropdown, a new field appears called "Time Sync Tolerance". In our Schema, this field has a default value of 30 - and thus we'd expect this to be the value in the field by default.

#### Actual behavior
In practice, the field is blank - and this is a general issue with these type of anyOf/oneOf dependent fields. Fixing this would be a great help for several real-life applications.

@epicfaace Hope the above is what you're looking for in regards to an example - else please let me know.

Best,
Martin

### PR DESCRIPTION
**Pull request by epicfaace**
### Reasons for making this change

Fixes #1293 - default with oneof/anyof
Fixes #1229, fixes #768 - default with nested dependencies

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by epicfaace** [comment]
Actually #1293 is not fixed completely yet, working on it

File: src/utils.js [hunk 1]
```diff
@@ -157,6 +157,16 @@ function computeDefaults(schema, parentDefaults, definitions = {}) {
     defaults = schema.items.map(itemSchema =>
       computeDefaults(itemSchema, undefined, definitions)
     );
+  } else if ("oneOf" in schema) {
+    schema =
+      schema["oneOf"][
```
Comment (LucianBuzzo) [inline src/utils.js]: NIT: This can just be `schema.oneOf`

File: src/utils.js [hunk 2]
```diff
@@ -157,6 +157,16 @@ function computeDefaults(schema, parentDefaults, definitions = {}) {
...
+        getMatchingOption(undefined, schema["oneOf"], definitions)
+      ];
+  } else if ("anyOf" in schema) {
+    schema =
+      schema["anyOf"][
```
Comment (LucianBuzzo) [inline src/utils.js]: NIT: This can just be `schema.anyOf`

File: test/utils_test.js [hunk 3]
```diff
@@ -315,6 +315,283 @@ describe("utils", () => {
...
+          },
+        };
+        expect(getDefaultFormState(schema, {})).eql({
+          name: {
+            first: "First Name",
+          },
+        });
+      });
+
+      it.only("should populate defaults for anyOf + dependencies", () => {
```
Comment (epicfaace) [inline test/utils_test.js]:
```suggestion
      it("should populate defaults for anyOf + dependencies", () => {
```

**Pull request merged**